### PR TITLE
steps: remove timemory variant / +codechecks enables -Werror

### PR DESCRIPTION
**Pull request by tristan0x**

### CODE REVIEW

**Review by matz-e — DISMISSED** [review]

**Review by pramodk — APPROVED** [review]
LGTM

File: deploy/environments/applications.yaml [hunk 1]
```diff
@@ -150,7 +150,7 @@ spack:
     - reportinglib%intel
     - spatial-index
     - spykfunc
-    - steps+petsc
+    - steps+distmesh+petsc
```
Comment (pramodk) [inline deploy/environments/applications.yaml]: @tristan0x : `distmesh` is only with `develop` right? As we are deploying `3.6.0` as a default, `distmesh` won't be deployed anyway?

**Review by pramodk — CHANGES_REQUESTED** [review]

File: deploy/environments/applications.yaml (hunk 1 above)
Comment (tristan0x) [reply deploy/environments/applications.yaml]: 3.6.0 is not affected by the +distmesh variant.
But spack CI seems to build the master branch of STEPS somehow, and the build of the master branch is broken without +distmesh  (I am fixing that).

**Comment by tristan0x** [comment]
won't merge this one.

**Pull request closed without merging**